projects/npm-tools/packages/npm-scripts/src/utils/createFederationConfig.js

```diff
@@ -189,6 +190,56 @@ module.exports = async function () {
 	remotes = remotes || [];
 	shared = shared || [];
 
+	// Prevent webpack from choking on scopes
+
+	const libraryName = makeNamespace(name);
+
+	const federationPluginConfig = {
+		exposes: {
+			...transformExposes(exposes, build.input),
+			'.': mainFilePath,
+		},
+		filename: 'container.js',
+		library: {
+			name: `window[Symbol.for("__LIFERAY_WEBPACK_CONTAINERS__")]["${libraryName}"]`,
+			type: 'assign',
+		},
+		name,
+		remoteType: 'script',
+		remotes: [...DEFAULT_REMOTES, ...remotes].reduce((remotes, remote) => {
+			let webContextPath;
+
+			if (typeof remote === 'string') {
+				webContextPath = remote;
+			} else {
+				webContextPath = remote.webContextPath;
+				remote = remote.name;
+			}
+
+			// Prevent webpack from choking on scopes
+
+			const libraryName = makeNamespace(name);
+
+			remotes[remote] =
+				`window[Symbol.for("__LIFERAY_WEBPACK_CONTAINERS__")]` +
+				`["${libraryName}"]` +
+				`@/o/${webContextPath}/__generated__/container.js`;
+
+			return remotes;
+		}, {}),
+		shared: [...DEFAULT_SHARED, ...shared].reduce((shared, name) => {
+			shared[name] = {singleton: true};
+
+			return shared;
+		}, {}),
+	};
+
+	createTempFile(
+		'federation.plugin.config.json',
+		JSON.stringify(federationPluginConfig, null, 2),
+		{autoDelete: false}
```

Comment:
I'm leaving this file in the build directory because I'm tired of commenting the deletion every time anything fails.

This will happen to anyone having to debug any problem with webpack federation, so better leave it there for anyone interested.